dspace-server-webapp/src/test/java/org/dspace/app/rest/ItemIdentifierControllerIT.java

```diff
@@ -0,0 +1,178 @@
+        context.turnOffAuthorisationSystem();
+
+        DOIService doiService = IdentifierServiceFactory.getInstance().getDOIService();
+
+        //** GIVEN **
+        //1. A community-collection structure with one parent community and two collections.
+        parentCommunity = CommunityBuilder.createCommunity(context)
+                .withName("Parent Community")
+                .build();
+        Collection col1 = CollectionBuilder.createCollection(context, parentCommunity).withName("Collection 1").build();
+
+        //2. A public item that is readable by Anonymous
+        Item publicItem1 = ItemBuilder.createItem(context, col1)
+                .withTitle("Public item 1")
+                .withIssueDate("2017-10-17")
+                .withAuthor("Smith, Donald")
+                .build();
+
+        // This item should not have a DOI
+        DOIIdentifierProvider doiIdentifierProvider =
+                DSpaceServicesFactory.getInstance().getServiceManager()
+                        .getServiceByName("org.dspace.identifier.DOIIdentifierProvider",
+                                org.dspace.identifier.DOIIdentifierProvider.class);
+        doiIdentifierProvider.delete(context, publicItem1);
+
+        // A non-admin should get an unauthorised error from REST method preauth
+        // Expect first forbidden
+        String token = getAuthToken(eperson.getEmail(), password);
+        getClient(token).perform(post("/api/core/items/" +
+                        publicItem1.getID().toString() + "/identifiers"))
+                .andExpect(status().isForbidden());
+
+        // Set token to admin credentials
+        token = getAuthToken(admin.getEmail(), password);
+
+        // Expect a successful 201 CREATED for this item with no DOI
+        getClient(token).perform(post("/api/core/items/" +
+                        publicItem1.getID().toString() + "/identifiers"))
+                .andExpect(status().isCreated());
+
+        // Expected 302 FOUND status code for a DOI already in REGISTERED / TO_BE_REGISTERED state
+        getClient(token).perform(post("/api/core/items/" +
+                        publicItem1.getID().toString() + "/identifiers"))
+                .andExpect(status().isFound());
+
+        // Get the doi we minted and queued for registration
+        DOI doi = doiService.findDOIByDSpaceObject(context, publicItem1);
+        // The DOI should not be null
+        assertNotNull(doi);
+        // The DOI status should be TO_BE_REGISTERED
+        Assert.assertEquals(DOIIdentifierProvider.TO_BE_REGISTERED, doi.getStatus());
+
+        // Now, set the DOI status back to pending and update
+        doi.setStatus(DOIIdentifierProvider.PENDING);
+        doiService.update(context, doi);
+
+        // Do another POST, again this should return 201 CREATED as we shift the DOI from PENDING to TO_BE_REGISTERED
+        getClient(token).perform(post("/api/core/items/" +
```

Comment:
Same here, this `POST` seems odd to me at a glance.  Why should an empty data POST result in a DOI moving from one status to the next?  It seems odd that the same endpoint which returns *all* identifiers when doing a `GET` can only perform actions on _DOIs_ when doing a `POST`.

Reply:
Yep, agreed as above, I can see how this isn't very clear.
DOI happens to be the only identifier type where we can support manual registration by an administrator. And my knowledge of REST and CRUD means POST == "Create" which to me is the kind of operation we're doing.

But we should still POST an identifier type. In the future, we might have a manual "Register ARK" button, and some identifier types might require additional data to be posted.

Reply:
I'm confused... how is this `POST /api/core/items/[uuid]/identifiers` still a valid command?  I thought this was moved to `POST /api/pid/identifiers`?

I think all the DOI creation tests should move to a new `IdentifierControllerIT` and use the new `/api/pid/identifiers` endpoint.

Reply:
we've resolvd this as part of the larger refactor